packages/ui-logic/src/error.ts

```diff
@@ -0,0 +1,30 @@
+import { Weave } from "./weave";
```

Comment:
you have circular dependency here.

Reply:
I'm not sure if I should include `weave.ts`'s code into `error.ts` so that it's not circular.

Reply:
The thing is `Weave`'s class may do other things besides parsing errors in the future.

Reply:
But you can move `ErrorParser.tryParseError` to the `wave.ts`. With different name of course.

Reply:
`ErrorParser.tryParseError` is a decorator of `Weave.tryParseError`, it parses other kind of errors besides Weave's ones. That's why I thought it may not fit in `weave.ts`.

Reply:
we need to avoid circular dependency somehow I think.

Reply:
I agree. How about having an `ErrorParser` class with `tryParseWeaveError` and `tryParseError`, and removing `weave.ts` since right now it does not do anything besides parsing weave errors?

Reply:
I think it is ok.